src/smartstream/derive/src/generator/filter.rs

```diff
@@ -0,0 +1,64 @@
+use quote::quote;
+use proc_macro2::TokenStream;
+use crate::SmartStreamFn;
+
+pub fn generate_filter_smartstream(func: &SmartStreamFn) -> TokenStream {
+    let user_fn = &func.name;
+    let user_code = func.func;
+
+    quote! {
+        #user_code
+
+        mod __system {
+            #[no_mangle]
+            #[allow(clippy::missing_safety_doc)]
+            pub unsafe fn filter(ptr: *mut u8, len: usize) -> i32 {
+                // DECODING
+                extern "C" {
+                    fn copy_records(putr: i32, len: i32);
+                }
+
+                let input_data = Vec::from_raw_parts(ptr, len, len);
+                let mut records: Vec<fluvio_smartstream::dataplane::record::Record> = vec![];
+                if let Err(_err) = fluvio_smartstream::dataplane::core::Decoder::decode(&mut records, &mut std::io::Cursor::new(input_data), 0) {
+                    return -1;
```

Comment:
why -1 ? at least define constant

Reply:
This is copy-pasted from your original filtering code :P I'll make a constant